### PR TITLE
Add an api to support ps/top.

### PR DESCRIPTION
**Pull request by rjnagal**
I need the ps/top output for the UI. Not sure if we should be adding it as an API. WDYT?

Also check the fields that I selected to show. Do we have any issues showing the full command. My guess is that we don't care.

### CODE REVIEW

File: manager/manager.go [hunk 1]
```diff
@@ -640,6 +643,27 @@ func (m *manager) Exists(containerName string) bool {
...
+	conts, err := m.getRequestedContainers(containerName, options)
+	if err != nil {
+		return nil, err
+	}
+	if len(conts) != 1 {
+		return nil, fmt.Errorf("Expected the request to match only one container")
+	}
+	// TODO(rjnagal): handle count? Only if we can do count by type (eg. top 5 cpu users)
+	ps := []v2.ProcessInfo{}
+	for _, cont := range conts {
```
Comment (vmarmol) [inline manager/manager.go]: nit: it's only one do we need the loop?

File: info/v2/container.go [hunk 2]
```diff
@@ -166,3 +166,16 @@ type RequestOptions struct {
...
 	Recursive bool `json:"recursive"`
 }
+
+type ProcessInfo struct {
+	User        string `json:"user"`
+	Pid         int    `json:"pid"`
+	Ppid        int    `json:"parent_pid"`
+	StartTime   string `json:"start_time"`
+	Pcpu        string `json:"percent_cpu"`
+	Rss         string `json:"rss"`
```
Comment (vmarmol) [inline info/v2/container.go]: nit: RSS and VirtualSize

File: info/v2/container.go [hunk 3]
```diff
@@ -166,3 +166,16 @@ type RequestOptions struct {
 	// Whether to include stats for child subcontainers.
 	Recursive bool `json:"recursive"`
 }
+
+type ProcessInfo struct {
+	User        string `json:"user"`
+	Pid         int    `json:"pid"`
+	Ppid        int    `json:"parent_pid"`
+	StartTime   string `json:"start_time"`
+	Pcpu        string `json:"percent_cpu"`
```
Comment (vmarmol) [inline info/v2/container.go]: nit: PercentCPU

File: container/raw/handler.go [hunk 4]
```diff
@@ -336,6 +336,10 @@ func (self *rawContainerHandler) GetStats() (*info.ContainerStats, error) {
 	return stats, nil
 }
 
+func (self *rawContainerHandler) GetProcesses() ([]int, error) {
```
Comment (vmarmol) [inline container/raw/handler.go]: Do we need this one? Why not ListProcesses()?

**Comment by vmarmol** [comment]
+1 to adding to UI/API and only querying on demand

File: info/v2/container.go [hunk 5]
```diff
@@ -166,3 +166,16 @@ type RequestOptions struct {
 	// Whether to include stats for child subcontainers.
 	Recursive bool `json:"recursive"`
 }
+
+type ProcessInfo struct {
+	User        string `json:"user"`
+	Pid         int    `json:"pid"`
+	Ppid        int    `json:"parent_pid"`
+	StartTime   string `json:"start_time"`
```
Comment (vmarmol) [inline info/v2/container.go]: Should we parse these values and get a more concrete API? Happy if we add a TODO :)

Comment (rjnagal) [inline info/v2/container.go]: I just kept the original strings, as I would have to convert them back to put on the UI :)
will update.

Comment (rjnagal) [inline info/v2/container.go]: Its actually convoluted to change the time fields back as the output changes based on the duration:

api/v2.0/ps/2ff16760cda11e8568589a9d73fe7e833b83ca3a8f55537157f7ff52195bad70?type=docker
[{"user":"root","pid":7003,"parent_pid":20768,"start_time":"May06","percent_cpu":"0.0","rss":"424","virtual_size":"7308","status":"Ss","running_time":"00:00:00","cmd":"stress"},{"user":"root","pid":7021,"parent_pid":7003,"start_time":"May06","percent_cpu":"99.5","rss":"262284","virtual_size":"269456","status":"R","running_time":"4-23:08:51","cmd":"stress"}]

start time can be May06, 20:11, or 2014. I think we can use a different option to print the whole time, but then that will be hard to condense back to be top like :/

Comment (vmarmol) [inline info/v2/container.go]: Should be fine for now then :)

File: manager/manager.go (hunk 1 above)
Comment (rjnagal) [inline manager/manager.go]: loop was simpler because conts is a map and containerName isn't necessarily the key.

Comment (vmarmol) [inline manager/manager.go]: aaah I didn't realize it was a map.

File: info/v2/container.go (hunk 3 above)
Comment (rjnagal) [inline info/v2/container.go]: done

File: container/raw/handler.go (hunk 4 above)
Comment (rjnagal) [inline container/raw/handler.go]: leftover from before I realized that we already had a ListProcesses() :) removed.

File: info/v2/container.go (hunk 2 above)
Comment (rjnagal) [inline info/v2/container.go]: done

File: api/versions.go [hunk 6]
```diff
@@ -391,6 +392,17 @@ func (self *version2_0) HandleRequest(requestType string, request []string, m ma
 		return writeResult(fi, w)
 	case eventsApi:
 		return handleEventRequest(request, m, w, r)
+	case psApi:
```
Comment (vmarmol) [inline api/versions.go]: Also add to SupportedRequestTypes(). Realized we can make this interface better and less manual :P

**Comment by vmarmol** [comment]
LGTM, just the nit. And run tests please :) since Travis is being mean.

File: api/versions.go (hunk 6 above)
Comment (rjnagal) [inline api/versions.go]: done

**Comment by vmarmol** [comment]
LGTM, thanks @rjnagal!

**Pull request merged**